scijava/scijava-ops/src/main/java/org/scijava/ops/OpInfo.java

```diff
@@ -29,9 +32,14 @@
 
 	/** Gets the associated {@link Struct} metadata. */
 	Struct struct();
-	
-	/** Describes whether this Op can be simplified. */
-	boolean isSimplifiable();
+
+	/** Gets the hints declared in the {@link OpHints} annotation */
+	List<String> declaredHints();
```

Comment:
OK, @gselzer: create a new `Hints` implementation that's read-only and backed by `ImmutableList`.

Reply:
Done!

Reply:
@gselzer I think we may want to return the actual `ImmutableHints` here, to concretely signal that the object is unmodifiable.  It's OK to have specific return types.

Reply:
@hinerm this is true, my only thought was that maybe there are times where it is too restrictive to return an `ImmutableHints`. But I can't think of a good use case, so I agree that we can make it `Immutable`

Reply:
@hinerm, ah, I forgot that the other reason for returning `Hints`: to return `ImmutableHints`, we'd have to `export` it. Is this something we want to do? I'll make a commit making this change so that I can move forward...

Reply:
@gselzer that's fair